### ISSUE TITLE
Not listening on 0.0.0.0

### ISSUE THREAD

**Issue by aslian** [nfx/slrp#21]
### Platform info
OS:
```
$ cat /etc/lsb-release /etc/upstream-release/lsb-release 
DISTRIB_ID=LinuxMint
DISTRIB_RELEASE=18.1
DISTRIB_CODENAME=serena
DISTRIB_DESCRIPTION="Linux Mint 18.1 Serena"
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04 LTS"
```

Kernel:
```
$ uname -a
Linux linux 4.4.0-53-generic #74-Ubuntu SMP Fri Dec 2 15:59:10 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

Binary from: https://github.com/nfx/slrp/releases/download/v0.0.7/slrp_0.0.7_linux_amd64.tar.gz

### Expected behaviour

When `addr: "0.0.0.0:8090"` is specified, listen on 0.0.0.0:8090 (ipv4).

### Observed behaviour
It start listening on IPv6 address and breaks

```
1638:5821  read(9</home/user/.slrp/config.yml>, "app:\n    state: $HOME/.slrp/data\n    sync: 1m\nlog:\n    level: warn\n    format: pretty\nserver:\n    addr: \"localhost:8089\"\n    read_timeout: 15s\n    enable_profiler: false\nmitm:\n    addr: \"0.0.0.0:8090\"\n    read_timeout: 15s\n    idle_timeout: 15s\n    write_timeout: 15s\nchecker:\n    timeout: 5s\n    strategy: simple\nhistory:\n    limit: 1000\n", 512) = 339
3555:5817  bind(9<TCPv6:[43840488]>, {sa_family=AF_INET6, sin6_port=htons(8090), inet_pton(AF_INET6, "::", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, 28 <unfinished ...>
3563:5817  epoll_ctl(4<anon_inode:[eventpoll]>, EPOLL_CTL_ADD, 9<TCPv6:[:::8090]>, {EPOLLIN|EPOLLOUT|EPOLLRDHUP|EPOLLET, {u32=1644746904, u64=140567334410392}} <unfinished ...>
3567:5817  getsockname(9<TCPv6:[:::8090]>,  <unfinished ...>
3569:5817  <... getsockname resumed> {sa_family=AF_INET6, sin6_port=htons(8090), inet_pton(AF_INET6, "::", &sin6_addr), sin6_flowinfo=0, sin6_scope_id=0}, [28]) = 0
3787:5816  accept4(9<TCPv6:[:::8090]>,  <unfinished ...>
```

```
$ netstat -tpln
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.1:8089          0.0.0.0:*               LISTEN      5814/slrp       
tcp6       0      0 :::8090                 :::*                    LISTEN      5814/slrp   
```

```
2022/08/19 19:43:01 http: panic serving 192.168.1.230:40620: As4 called on IPv6 address
goroutine 8668 [running]:
net/http.(*conn).serve.func1()
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:1825 +0xbf
panic({0xb0dd60, 0xeeab60})
	/opt/hostedtoolcache/go/1.18.5/x64/src/runtime/panic.go:844 +0x258
net/netip.Addr.As4({{0x9?, 0xc038d8?}, 0xc0001a40d8?})
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/netip/netip.go:699 +0x8e
github.com/nfx/slrp/pmux.NewProxy({0xc003469800?, 0x1f4?}, {0xc038d8, 0x4})
	/home/runner/work/slrp/slrp/pmux/proxy.go:224 +0x85
github.com/nfx/slrp/pmux.HttpProxy(...)
	/home/runner/work/slrp/slrp/pmux/proxy.go:237
github.com/nfx/slrp/serve.(*HttpProxyServer).Proxy(0xef08d8?)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:64 +0x45
github.com/nfx/slrp/serve.(*HttpProxyServer).handleConnect(0xc0001cc480, {0xef2110, 0xc00339a000}, 0xc000d08200)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:118 +0x3a5
github.com/nfx/slrp/serve.(*HttpProxyServer).ServeHTTP(0xef2840?, {0xef2110?, 0xc00339a000?}, 0x14f8930?)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:84 +0x54
github.com/nfx/slrp/serve.(*MitmProxyServer).ServeHTTP(0xc0001cc480, {0xef2110, 0xc00339a000}, 0xc000d08100)
	/home/runner/work/slrp/slrp/serve/mitm_proxy.go:50 +0x52b
net/http.serverHandler.ServeHTTP({0xc00117aa20?}, {0xef2110, 0xc00339a000}, 0xc000d08100)
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:2916 +0x43b
net/http.(*conn).serve(0xc0027a65a0, {0xef28e8, 0xc00011d0e0})
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:1966 +0x5d7
created by net/http.(*Server).Serve
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:3071 +0x4db
2022/08/19 19:43:01 http: panic serving 192.168.1.230:40624: As4 called on IPv6 address
goroutine 8618 [running]:
net/http.(*conn).serve.func1()
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:1825 +0xbf
panic({0xb0dd60, 0xeeab60})
	/opt/hostedtoolcache/go/1.18.5/x64/src/runtime/panic.go:844 +0x258
net/netip.Addr.As4({{0x9?, 0xc038d8?}, 0xc0001a40d8?})
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/netip/netip.go:699 +0x8e
github.com/nfx/slrp/pmux.NewProxy({0xc00375f120?, 0x1f4?}, {0xc038d8, 0x4})
	/home/runner/work/slrp/slrp/pmux/proxy.go:224 +0x85
github.com/nfx/slrp/pmux.HttpProxy(...)
	/home/runner/work/slrp/slrp/pmux/proxy.go:237
github.com/nfx/slrp/serve.(*HttpProxyServer).Proxy(0xef08d8?)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:64 +0x45
github.com/nfx/slrp/serve.(*HttpProxyServer).handleConnect(0xc0001cc480, {0xef2110, 0xc0000c8000}, 0xc0014a9a00)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:118 +0x3a5
github.com/nfx/slrp/serve.(*HttpProxyServer).ServeHTTP(0xef2840?, {0xef2110?, 0xc0000c8000?}, 0x14f8930?)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:84 +0x54
github.com/nfx/slrp/serve.(*MitmProxyServer).ServeHTTP(0xc0001cc480, {0xef2110, 0xc0000c8000}, 0xc0014a9500)
	/home/runner/work/slrp/slrp/serve/mitm_proxy.go:50 +0x52b
net/http.serverHandler.ServeHTTP({0xc00132b320?}, {0xef2110, 0xc0000c8000}, 0xc0014a9500)
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:2916 +0x43b
net/http.(*conn).serve(0xc0001ad400, {0xef28e8, 0xc00011d0e0})
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:1966 +0x5d7
created by net/http.(*Server).Serve
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:3071 +0x4db
2022/08/19 19:43:01 http: panic serving 192.168.1.230:40626: As4 called on IPv6 address
goroutine 8669 [running]:
net/http.(*conn).serve.func1()
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:1825 +0xbf
panic({0xb0dd60, 0xeeab60})
	/opt/hostedtoolcache/go/1.18.5/x64/src/runtime/panic.go:844 +0x258
net/netip.Addr.As4({{0x9?, 0xc038d8?}, 0xc0001a40d8?})
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/netip/netip.go:699 +0x8e
github.com/nfx/slrp/pmux.NewProxy({0xc00375f140?, 0x1f4?}, {0xc038d8, 0x4})
	/home/runner/work/slrp/slrp/pmux/proxy.go:224 +0x85
github.com/nfx/slrp/pmux.HttpProxy(...)
	/home/runner/work/slrp/slrp/pmux/proxy.go:237
github.com/nfx/slrp/serve.(*HttpProxyServer).Proxy(0xef08d8?)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:64 +0x45
github.com/nfx/slrp/serve.(*HttpProxyServer).handleConnect(0xc0001cc480, {0xef2110, 0xc0000c80e0}, 0xc0014a9b00)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:118 +0x3a5
github.com/nfx/slrp/serve.(*HttpProxyServer).ServeHTTP(0xef2840?, {0xef2110?, 0xc0000c80e0?}, 0x14f8930?)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:84 +0x54
github.com/nfx/slrp/serve.(*MitmProxyServer).ServeHTTP(0xc0001cc480, {0xef2110, 0xc0000c80e0}, 0xc0000bf900)
	/home/runner/work/slrp/slrp/serve/mitm_proxy.go:50 +0x52b
net/http.serverHandler.ServeHTTP({0xc001ead8f0?}, {0xef2110, 0xc0000c80e0}, 0xc0000bf900)
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:2916 +0x43b
net/http.(*conn).serve(0xc0027a66e0, {0xef28e8, 0xc00011d0e0})
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:1966 +0x5d7
created by net/http.(*Server).Serve
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:3071 +0x4db
2022/08/19 19:43:01 http: panic serving 192.168.1.230:40628: As4 called on IPv6 address
goroutine 8670 [running]:
net/http.(*conn).serve.func1()
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:1825 +0xbf
panic({0xb0dd60, 0xeeab60})
	/opt/hostedtoolcache/go/1.18.5/x64/src/runtime/panic.go:844 +0x258
net/netip.Addr.As4({{0x9?, 0xc038d8?}, 0xc0001a40d8?})
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/netip/netip.go:699 +0x8e
github.com/nfx/slrp/pmux.NewProxy({0xc00375f164?, 0x1f4?}, {0xc038d8, 0x4})
	/home/runner/work/slrp/slrp/pmux/proxy.go:224 +0x85
github.com/nfx/slrp/pmux.HttpProxy(...)
	/home/runner/work/slrp/slrp/pmux/proxy.go:237
github.com/nfx/slrp/serve.(*HttpProxyServer).Proxy(0xef08d8?)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:64 +0x45
github.com/nfx/slrp/serve.(*HttpProxyServer).handleConnect(0xc0001cc480, {0xef2110, 0xc0000c8460}, 0xc0014a9d00)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:118 +0x3a5
github.com/nfx/slrp/serve.(*HttpProxyServer).ServeHTTP(0xef2840?, {0xef2110?, 0xc0000c8460?}, 0x14f8930?)
	/home/runner/work/slrp/slrp/serve/http_proxy.go:84 +0x54
github.com/nfx/slrp/serve.(*MitmProxyServer).ServeHTTP(0xc0001cc480, {0xef2110, 0xc0000c8460}, 0xc0014a9c00)
	/home/runner/work/slrp/slrp/serve/mitm_proxy.go:50 +0x52b
net/http.serverHandler.ServeHTTP({0xc00132bad0?}, {0xef2110, 0xc0000c8460}, 0xc0014a9c00)
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:2916 +0x43b
net/http.(*conn).serve(0xc0027a6780, {0xef28e8, 0xc00011d0e0})
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:1966 +0x5d7
created by net/http.(*Server).Serve
	/opt/hostedtoolcache/go/1.18.5/x64/src/net/http/server.go:3071 +0x4db
```

**Comment by nfx** [comment]
@aslian interesting. did you try specifying the address of the interface? e.g. i don't want this app to listen on `0.0.0.0` for security reasons and want it to pick up internally-routable addresses only. Ideally picking that up automatically, eventually.

**Comment by nfx** [comment]
issue was related to logging behaviour. will be released in v0.0.8

@aslian  please open a new issue if this behaviour repeats. 

P.S. `./slrp -update` makes it easy ;)
